posthog/models/hog_functions/hog_function.py

```diff
@@ -0,0 +1,115 @@
+import json
+from typing import Optional
+
+from django.db import models
+from django.db.models.signals import post_save
+from django.dispatch.dispatcher import receiver
+
+from posthog.models.action.action import Action
+from posthog.models.team.team import Team
+from posthog.models.utils import UUIDModel
+from posthog.redis import get_client
+
+
+class HogFunction(UUIDModel):
```

Comment:
Re naming things: I think `HogFunction` is fine, but like I said once before, in many ways these are full programs, not just functions, so I wonder if `HogProgram` or `HogApp` or `HogScript` will be a better name, though overall this is very low on my "give a f..." scale, so happy with whatever.